packages/story-utils/src/getStories.ts

```diff
@@ -10,13 +11,21 @@ type StoryData = {
   storyFn: Story<unknown>;
 };
 
+const metadataSchema = zod
+  .object({
+    title: zod.string(),
+    parameters: zod.record(zod.unknown()).optional(),
+  })
+  .nonstrict();
+
 export default function getStories(globPath = '**/*.stories.*'): StoryData[] {
   const filePaths = glob.sync(globPath);
 
   return filePaths.flatMap((filePath) => {
     // eslint-disable-next-line @typescript-eslint/no-var-requires
     const storiesFileExports = require(path.resolve(filePath));
-    const { default: metadata, ...storiesMap } = storiesFileExports;
+    const { default: rawMetadata, ...storiesMap } = storiesFileExports;
+    const metadata = metadataSchema.parse(rawMetadata);
```

Comment:
The `metadata` variable now has good typescript typing. And the `parse` step will blow up if `rawMetadata` doesn't match the schema.